src/bosh.js

```diff
@@ -78,6 +78,7 @@ Strophe.Request.prototype = {
      *
      *  Throws:
      *    "parsererror" - A parser error occured.
+     *    "badformat" - The entity has sent XML that cannot be processed.
      *
```

Comment:
From your [documentation](http://strophe.im/strophejs/doc/1.2.3/files/strophe-js.html#Strophe.Connection.connect):

> The error condition will be one of the conditions defined in RFC 3920 or the condition ‘strophe-parsererror’.

Therefore I used the best fitting condition from [RFC 3920](https://tools.ietf.org/html/rfc3920#section-4.7.3):

> &lt;bad-format/> -- the entity has sent XML that cannot be processed;

I think we should stick to the RFC as close as possible, or not?